File: ios/Classes/LogLevelPlugin.swift
import Flutter
import UIKit
import SuperwallKit

public class LogLevelPlugin: NSObject, FlutterPlugin {
  public static func register(with registrar: FlutterPluginRegistrar) {
    let channel = FlutterMethodChannel(name: "SWK_LogLevelPlugin", binaryMessenger: registrar.messenger())
    let instance = LogLevelPlugin()
    registrar.addMethodCallDelegate(instance, channel: channel)
  }
  
  public func handle(_ call: FlutterMethodCall, result: @escaping FlutterResult) {
    guard let args = call.arguments as? [String: Any],
          let levelValue = args["level"] as? Int,
          let logLevel = LogLevel(rawValue: levelValue) else {
      result(FlutterError(code: "INVALID_ARGUMENTS", message: "Invalid log level", details: nil))
      return
    }

    switch call.method {
      case "getLogLevelDescription":
        result(logLevel.description)
      case "getLogLevelDescriptionEmoji":
        result(logLevel.descriptionEmoji)
      default:
        result(FlutterMethodNotImplemented)
    }
  }
}

File: android/src/main/kotlin/com/superwall/superwallkit_flutter/LogLevelPlugin.kt
package com.superwall.superwallkit_flutter

import LogLevel
import io.flutter.embedding.engine.plugins.FlutterPlugin
import io.flutter.plugin.common.MethodCall
import io.flutter.plugin.common.MethodChannel
import io.flutter.plugin.common.MethodChannel.MethodCallHandler
import io.flutter.plugin.common.MethodChannel.Result

class LogLevelPlugin : FlutterPlugin, MethodCallHandler {

    private lateinit var channel: MethodChannel

    override fun onAttachedToEngine(flutterPluginBinding: FlutterPlugin.FlutterPluginBinding) {
        channel = MethodChannel(flutterPluginBinding.binaryMessenger, "SWK_LogLevelBridge")
        channel.setMethodCallHandler(this)
    }

    override fun onMethodCall(call: MethodCall, result: Result) {
        val args = call.arguments<Map<String, Any>?>()
        val levelValue = args?.get("level") as? Int
        val logLevel = levelValue?.let { findLogLevel(it) }

        if (logLevel != null) {
            when (call.method) {
                "getLogLevelDescription" -> result.success(logLevel.toString())
                "getLogLevelDescriptionEmoji" -> result.success(logLevel.getDescriptionEmoji())
                else -> result.notImplemented()
            }
        } else {
            result.error("INVALID_ARGUMENTS", "Invalid log level", null)
        }
    }

    private fun findLogLevel(value: Int): LogLevel? {
        return LogLevel.values().firstOrNull { it.level == value }
    }

    override fun onDetachedFromEngine(binding: FlutterPlugin.FlutterPluginBinding) {
        channel.setMethodCallHandler(null)
    }
}
